bench_test.go

```diff
@@ -54,6 +54,7 @@ func BenchmarkSendRecv(b *testing.B) {
 	recvBuf := make([]byte, 512)
 
 	doneCh := make(chan struct{})
+	b.ResetTimer()
```

Comment:
Suggest you also add `b.ReportAllocs()` here and in `BenchmarkSendRecvLarge`

Reply:
I am not sure this is necessary, you can add `-benchmem` to `go test` to achieve the same result.

Reply:
True, but I'm suggesting it because allocs seem to be of ongoing concern. Up to you.